lib/node_modules/@stdlib/utils/parse-dsv/lib/parse_dsv.js

```diff
@@ -0,0 +1,210 @@
+const fs = require('fs');
+const Buffer = require('buffer').Buffer;
+
+const DQUOTE = '"';
+const CR = '\r';
+const LF = '\n';
+const CRLF = '\r\n';
+
+const token_enum = {
+    'DELIM': 0,
+    'DQUOTE': 1,
+    'CR': 2,
+    'LF': 3,
+    'CRLF': 4,
+    'TEXTDATA': 5
+}
+
+const state_enum = {
+    'INITIAL': 0,
+    'REST': 1,
+    'RECORD': 2,
+    'RECORDREST': 3,
+    'FIELD': 4,
+    'ESCAPED': 5,
+    'NONESCAPED': 6,
+    'TWODQUOTE': 7
+}
+
+function parse_dsv_stream(DELIM = ',') {
+
+    let csv_state = [];
+    let buffer = [];
+    let state = state_enum.INITIAL;
+    let row = -1;
+    let column = -1;
+
+    return incrparse;
+
+    // bytes should be an array
+    // an array of characters (strings with length 1)
+    function incrparse(chars) {
```

Comment:
Would it make more sense for `chars` to be either a `string` or `Buffer`/`Uint8Array`?  I am thinking from the standpoint of reading data from disk or when wrapped in a stream, etc.

Reply:
Otherwise, for something like reading data from disk, you need to `buf.toString().split( '' )` in order to feed data to the accumulator.

Reply:
I actually do want the inner buffer variable to be an instance of Buffer instead of []. I've only been using a normal array to take advantage of shift/unshift/splice methods for an easier time implementing things.

It might bring up issues with UTF-8 though.

Reply:
@rei2hu Re: shift/unshift operations. Ah, right. Makes sense. My point was more that the values/data provided to `incrparse` can be a string/Buffer/Uint8Array, similar to the `fs` and `stream` methods found in Node.js. The internal `buffer` variable being a dynamic array is a reasonable choice, I think.

Reply:
I meant to say if I changed the dynamic array to an instance of Buffer, I would be able to use `buffer.concat` which avoids having to transform the buffer to something else.

Although just transforming the provided buffer to something that can be added onto the end of a dynamic array is a lot simpler so I'll probably go with that for now.